src/hotspot/share/runtime/objectMonitor.cpp

```diff
@@ -316,13 +317,13 @@ oop ObjectMonitor::object() const {
 void ObjectMonitor::set_object_strong() {
   check_object_context();
   if (_object_strong.is_empty()) {
-    if (AtomicAccess::cmpxchg(&_object_strong_lock, 0, 1) == 0) {
+    auto setObjectStrongLambda = [&](OopHandle& object_strong, const WeakHandle& object) {
```

Comment:
I don't understand why we need the complexity of the `SpinSingleSection` and use of lambda's/functors.

This seems like try-lock usage, though I'm not at all sure why (i.e. if we don't get the lock who is taking care of making a strong reference?)